### DISCUSSION TITLE
Square Thumbnails?

### DISCUSSION THREAD

**Discussion by eecue** [RhetTbull/osxphotos#904] [General]
I'm planning on exporting square thumbnails of my images. I'm wondering if there's some data in Photos that I can use to detect a good point of interest to crop on.

**Comment by RhetTbull** [comment]
That's a great question.  As far as I know, such data is not present in the Photos database and if it is, osxphotos does not expose it. I took another look at the [schema](https://github.com/RhetTbull/osxphotos/wiki/Photos-8-schema-(Ventura---MacOS-13.0.0)) and nothing jumps out at me as a possible candidate.  For photos with detected faces, the centroids of those faces are recorded (and osxphotos makes these available via the [FaceInfo](https://github.com/RhetTbull/osxphotos/blob/master/API_README.md#faceinfo) class).

**Reply by RhetTbull** [reply]
Thinking some more about this, a relatively simple approach would be:

- If photo has faces in it, try to avoid cropping a face -- find bounding box of each face and try to crop so faces aren't cut off. If not possible, place maximum number of faces towards center of the square (or find centroid of the group of face centroids (e.g. the center of the group) and crop around that.

- If photo doesn't have faces, use the centroid of the photo

This is pretty simplistic but might be good enough. I'm sure it's possible to use ML to identify the "most pleasing" area to crop (Photos already computes [aesthetic scores](https://github.com/RhetTbull/osxphotos/blob/master/API_README.md#scoreinfo) for each photo) but I'm not sure how to go about this myself.

**Reply by eecue** [reply]
Yeah I think that's for sure good enough.  Thanks!
